wgsl/syntax.bnf

```diff
@@ -579,18 +579,18 @@ enable_extension_list :
 ;
 
 requires_directive :
-  'requires' software_extension_list ';'
+  'requires' language_extension_list ';'
```

Comment:
@dneto0 Thank you very much for the draft. These diffs should not appear as they are already merged, a rebase/update against main would remove them. Thank you very much in advance for consideration.

Reply:
I was making this PR when I noticed that naming thing.  Then I make the renaming PR, and came back to this one.

I'm using the jj but am still getting used to it. I think I've properly rebased it now.

This is still a draft, so feel free to look around, but I'm not pushing this at committee yet. I haven't showed it to colleagues.  I think it's in pretty good shape, however.

Reply:
@dneto0 thank you! I like that swizzles are not a new/separate syntax rule but name (as they are valid names) in the grammar. Swizzles will make it easier to translate from other shader languages & vector math libraries, thanks a lot for the draft.